app/src/main/java/io/github/droidkaigi/confsched2017/view/customview/ParticlesAnimationView.java

```diff
@@ -100,68 +122,43 @@ public Shader resize(int width, int height) {
     private List<Particle> createParticles(int count) {
         List<Particle> particles = new ArrayList<>();
         for (int i = 0; i < count; i++) {
-            particles.add(new Particle(getWidth(), getHeight()));
+            particles.add(new Particle(getWidth(), getHeight(), this));
         }
         return particles;
     }
 
-    private List<Line> createLines(List<Particle> particles) {
-        final List<Line> lines = new ArrayList<>();
-        Stream.of(particles)
-                .forEach(particle -> Stream.of(particles)
-                        .filter(targetParticle ->
-                                particle.isShouldLinked(LINK_HEXAGON_DISTANCE, targetParticle)
-                        )
-                        .map(linkParticle -> new Line(new float[]{
-                                particle.center.x,
-                                particle.center.y,
-                                linkParticle.center.x,
-                                linkParticle.center.y})
-                        )
-                        .forEach(line -> {
-                            if (!lines.contains(line)) lines.add(line);
-                        })
-                );
-        return lines;
-    }
-
-    private class Line {
+    /**
+     * A Pair of 'Particles' whose centers can be 'linked to each other ... Called 'Line'.
+     */
+    private static class Line extends Pair<Particle, Particle> {
 
         private static final int MAX_ALPHA = 172;
-        private float[] point;
 
-        public Line(float[] point) {
-            this.point = point;
+        /**
+         * Constructor for a Pair.
+         *
+         * @param first  the first object in the Pair
+         * @param second the second object in the pair
+         */
+        Line(Particle first, Particle second) {
+            super(first, second);
         }
 
-        public void draw(Canvas canvas, Paint paint) {
-            final double distance = Math.floor(Math.sqrt(
-                    (point[2] - point[0]) * (point[2] - point[0])
-                            + (point[3] - point[1]) * (point[3] - point[1])
-            ));
+        void draw(Canvas canvas, Paint paint) {
+            if (!first.shouldBeLinked(LINK_HEXAGON_DISTANCE, second)) {
```

Comment:
In my opinion, `shouldBeLinked` is not necessary if first particle compare second particle in `draw`.